Engine/EditorUI/Panel/PanelMenuBar.cpp

```diff
@@ -99,21 +103,32 @@ void PanelMenuBar::ShowGameObjectMenu()
 		
 		if (ImGui::BeginMenu("3D Object"))
 		{
+			GameObject* created_game_object = nullptr;
+
 			if (ImGui::Selectable("Cube"))
 			{
-				App->model_loader->LoadCoreModel(PRIMITIVE_CUBE_PATH);
+				created_game_object = TemplatedGameObjectCreator::CreatePrimitive(CoreResource::CUBE);
 			}
 			if (ImGui::Selectable("Cylinder"))
 			{
-				App->model_loader->LoadCoreModel(PRIMITIVE_CYLINDER_PATH);
+				created_game_object = TemplatedGameObjectCreator::CreatePrimitive(CoreResource::CYLINDER);
 			}
 			if (ImGui::Selectable("Sphere"))
 			{
-				App->model_loader->LoadCoreModel(PRIMITIVE_SPHERE_PATH);
+				created_game_object = TemplatedGameObjectCreator::CreatePrimitive(CoreResource::SPHERE);
 			}
 			if (ImGui::Selectable("Torus"))
 			{
-				App->model_loader->LoadCoreModel(PRIMITIVE_TORUS_PATH);
+				created_game_object = TemplatedGameObjectCreator::CreatePrimitive(CoreResource::TORUS);
+			}
+			if (ImGui::Selectable("Quad"))
+			{
+				created_game_object = TemplatedGameObjectCreator::CreatePrimitive(CoreResource::QUAD);
```

Comment:
this code repeats itself in Panel Hierarchy, why? Maybe put this piece of code into a function

Reply:
Yeah, thats right. I was thinking about making our own hpp file with definitions of useful `imgui` code.